public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import {
+  KeyValuePair,
+  Link,
+  Span,
+  Trace,
+  TraceTimelineViewer,
+  TTraceTimeline,
+  UIElementsContext,
+  transformTraceData,
+  SpanData,
+  TraceData,
+  TracePageHeader,
+} from '@jaegertracing/jaeger-ui-components';
+import { UIElements } from './uiElements';
+import { useViewRange } from './useViewRange';
+import { useSearch } from './useSearch';
+import { useChildrenState } from './useChildrenState';
+import { useDetailState } from './useDetailState';
+import { useHoverIndentGuide } from './useHoverIndentGuide';
+
+type Props = {
+  trace: TraceData & { spans: SpanData[] };
+};
+
+export function TraceView(props: Props) {
```

Comment:
Part of refactoring was that hooks went into separate files and grouped into specialized custom hooks

Reply:
Nice!